pkg/kv/kvserver/metrics.go

```diff
@@ -96,6 +97,25 @@ var (
 		Unit:        metric.Unit_COUNT,
 	}
 
+	metaRaftFlowsReplicate = metric.Metadata{
+		Name:        "raft.flows.state-replicate",
+		Help:        "Number of leader->peer flows in StateReplicate",
+		Measurement: "Flows",
+		Unit:        metric.Unit_COUNT,
+	}
+	metaRaftFlowsProbe = metric.Metadata{
+		Name:        "raft.flows.state-probe",
```

Comment:
Any reason to use `-` instead of `_` for `state-x`?

iirc, `.` changes into `_` in prometheus, is that why?

Reply:
No particular reason, just instinctively typed it like this. I guess I use `-` as a more tight coupling sign, and `_` if these are completely "separate" terms. Looking at this file, both `_` and `-` are used (e.g. a bunch of `rocksdb`/`storage` metrics), so I'm not sure what the convention is.

Reply:
Is `-` converted to `_` in prometheus? In this case maybe going for `_` is better so that this is consistent and searchable.

Reply:
I think that `-` stays the same, you could check `status/vars` endpoint to check what prometheus is scraping.

Reply:
I think there is another layer of obfuscation somewhere, e.g. the `rocksdb.memtable.total-size` metric becomes `rocksdb_memtable_total_size`:

<img width="719" alt="Screenshot 2024-12-10 at 13 21 20" src="https://github.com/user-attachments/assets/7ce14a6b-ddac-47e3-8d11-f17609a9fb11">

Switched to `_` to minimize the diff.

Reply:
Ah I forgot that. That sound good.